src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +81,42 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+  
+  /** Stores the User in Datastore. */
+  public void storeUser(User user) {
+   Entity userEntity = new Entity("User", user.getEmail());
+   userEntity.setProperty("email", user.getEmail());
+   userEntity.setProperty("aboutMe", user.getAboutMe());
+   datastore.put(userEntity);
+  }
+  
+  /**
+   * Returns the User owned by the email address, or
+   * null if no matching User was found.
+   */
+  public User getUser(String email) {
+  
+   Query query = new Query("User")
+     .setFilter(new Query.FilterPredicate("email", FilterOperator.EQUAL, email));
+   PreparedQuery results = datastore.prepare(query);
+   Entity userEntity = results.asSingleEntity();
+   if(userEntity == null) {
+    return null;
+   }
+   
+   String aboutMe = (String) userEntity.getProperty("aboutMe");
+   User user = new User(email, aboutMe);
+   
+   return user;
+  }
+  
+  public Set<String> getUsers(){
+	  Set<String> users = new HashSet<>();
+	  Query query = new Query("Message");
```

Comment:
It looks like here you are querying messages instead of users here. The current code would query all the messages and grab the associated users. 

You can query the users like below:
```suggestion
	  Query query = new Query("User");
```

Reply:
I see what you mean.  Thank you for your suggestion!